middleware/cors/cors_test.go

```diff
@@ -689,6 +689,43 @@ func Test_CORS_AllowCredentials(t *testing.T) {
 	}
 }
 
+// The Enhancement for issue #2804
+func Test_CORS_AllowPrivateNetworkAccess(t *testing.T) {
```

Comment:
We should add a test were the flag is set to False, and make sure the Header is not present.